configuration/pih/liquibase/liquibase.xml

```diff
@@ -144,7 +144,36 @@
             drop table if exists temp_value_drug;
         </sql>
     </changeSet>
-
+<changeSet id="20230201-update-programs-for-dead-patients-aali" author="aali">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+               select CASE WHEN count(*) > 0 THEN 1 ELSE 0 END AS cnt
+               from patient_program pp
+               inner join program p2 on p2.program_id = pp.program_id
+               where voided = 0
+               and date_completed  is null
+               and EXISTS ( select 1 from person p where p.person_id = pp.patient_id and p.dead = 1); 
+            </sqlCheck>
+        </preConditions>
+        <comment>If the count is 0, then remove these programs (Hypertension program, Diabetes program, Epilepsy program)</comment>
+        <sql>
+         UPDATE patient_program pp2, (
+            SELECT patient_id , max(encounter_datetime) AS encounter_date
+            FROM encounter e WHERE patient_id IN (
+               select pp.patient_id
+               from patient_program pp
+               inner join program p2 on p2.program_id = pp.program_id
+               where voided = 0
+               and date_completed  is null
+               and EXISTS
+                ( select 1 from person p where p.person_id = pp.patient_id
+                and p.dead = 1)
+            )
```

Comment:
I only see references to the max(encounter_date) here, no references to death date... don't we want to use death date first, and only fall back to most recent encounter date if the patient doesn't have a death date?

Reply:
The main issue was to close programs for the dead patients, so I am looking into dead patinets and they have open programs, then update program close date with the date of the last encounter for each patient.

Reply:
@dmdesimone  just wanted to confirm that the above logic works for you, since it differs from the ticket description (which says to use death date as the date completed)

Reply:
Yes, that's what Kasseem confirmed.  I updated the description.

Reply:
Sounds good, thanks @dmdesimone  and @aibrahem550 !